packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,271 @@
+              scrollOffset:
+                  scrollOffset + model.last.maxSliver + crossAxisSpacing,
+            ),
+          );
+          break;
+      }
+
+      return DynamicSliverGridGeometry(
+        scrollOffset: model.last.scrollOffset,
+        crossAxisOffset: 0.0,
+        mainAxisExtent: childSize.height + mainAxisSpacing,
+        crossAxisExtent: childSize.width + crossAxisSpacing,
+      );
+    } else {
+      model.last.currentSizeUsed = addedSize;
+    }
+
+    switch (scrollDirection) {
+      case Axis.vertical:
+        if (childSize.height + mainAxisSpacing > model.last.maxSliver) {
+          model.last.maxSliver = childSize.height + mainAxisSpacing;
+        }
+        break;
+      case Axis.horizontal:
+        if (childSize.width + crossAxisSpacing > model.last.maxSliver) {
+          model.last.maxSliver = childSize.width + crossAxisSpacing;
+        }
+        break;
+    }
+
+    return DynamicSliverGridGeometry(
+      scrollOffset: scrollOffset,
+      crossAxisOffset: currentSizeUsed,
+      mainAxisExtent: childSize.height,
+      crossAxisExtent: childSize.width,
+    );
+  }
+
+  @override
+  bool reachedTargetScrollOffset(double targetOffset) {
+    if (model.last.scrollOffset > targetOffset) {
+      return true;
+    }
+    return false;
+  }
+}
+
+// TODO(snat-s): YourWrapDelegate extends SliverGridDelegate
+
+/// A [SliverGridDelegate] that makes it possible to create grids that wrap and
+/// have variably sized tiles.
+///
+/// For example, if the grid is vertical, this delegate will create a layout
+/// where the children are layed out until they fill the horizontal axis and then
+/// they pass to the second row. If the grid is horizontal, this delegate will
+/// do the same but it will fill the vertical axis and will pass to another
+/// column until it finishes.
+///
+/// This delegate creates grids with different sized tiles. But you
+/// can have equally sized tiles if you modify the `childCrossAxisExtent` and
+/// the `childMainAxisExtent` to be the same size.
+///
+/// See also:
+///  * [DynamicGridView.wrap], wich is a constructor to use this [SliverGridDelegate],
+///    like `GridView.extent`.
+///  * [DynamicGridView], which can use this delegate to control the layout of its
+///    tiles.
+///  * RenderDynamicSliverGrid], which can use this delegate to control the
+///    layout of its tiles.
+class SliverGridDelegateWithWrapping extends SliverGridDelegate {
+  /// Creates a delegate that makes dynamic grid layouts, the children Widgets
+  /// are the ones that decide their own size. Because the children decide their
+  /// own size the default values of `childMainAxisExtent` and
+  /// `childCrossAxisExtent` are set to [double.infinity].
+  /// If you only want to have the wrap, you can specify the `childCrossAxisExtent`
+  /// and the `childMainAxisExtent` to be the same size. Or only one of them to
+  /// be of a certain size in one of the axis.
```

Comment:
This also might be a good place to clarify that the child does not decide its size, but rather it is provided loose constraints, and if the extent parameters are set, the children are given tight constraints